### PR TITLE
enhancement: polish Auction Results for you screen

### PR DESCRIPTION
**Pull request by olerichter00**
The type of this PR is: Enhancement

This PR resolves [CX-1659]

### Description

- Adds '/auction-results-for-you' to android manifest
- Removes unnecessary artist slug from query
- Cleans up some styling
- Fixes auction results grouping bug
- Gives auction results section header a white background (it's not transparent anymore)
- Fixes auction results list item key extractor
- Changes title of auction results home rail to "Auction Results for Artists You Follow"
- Enables sticky section headers for Android

https://user-images.githubusercontent.com/4691889/123801813-e9b78f00-d8ea-11eb-9a6a-efd3761f332a.mp4

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Polish Auction Results for you screen and home rail (behind feature flag) - ole

[CX-1659]: https://artsyproduct.atlassian.net/browse/CX-1659

### CODE REVIEW

File: src/lib/Scenes/AuctionResultsForYou/AuctionResultsForYou.tsx [hunk 1]
```diff
@@ -28,10 +27,13 @@ interface Props {
 export const AuctionResultsForYou: React.FC<Props> = ({ me, relay }) => {
   const { hasMore, isLoading, loadMore } = relay
   const [loadingMoreData, setLoadingMoreData] = useState(false)
-  const auctionResultData = extractNodes(me?.auctionResultsByFollowedArtists)
-  const groupedAuctionResultData = groupBy(auctionResultData, (item) => moment(item!.saleDate!).format("MMM"))
-  const groupedAuctionResultSectionData = Object.keys(groupedAuctionResultData).map((key) => {
-    return { date: key, data: groupedAuctionResultData[key] }
+  const allAuctionResults = extractNodes(me?.auctionResultsByFollowedArtists)
+  const groupedAuctionResults = groupBy(allAuctionResults, (item) => moment(item!.saleDate!).format("YYYY-MM"))
```
Comment (olerichter00) [inline src/lib/Scenes/AuctionResultsForYou/AuctionResultsForYou.tsx:31]: Note: We need to group by month and year (e.g. `2021-05`) and then extract the section title (e.g. `January`).

File: src/lib/Scenes/AuctionResultsForYou/AuctionResultsForYou.tsx [hunk 2]
```diff
@@ -51,42 +53,48 @@ export const AuctionResultsForYou: React.FC<Props> = ({ me, relay }) => {
...
-          renderSectionHeader={({ section: { date } }) => (
-            <Text
-              textAlign="left"
-              color="black"
-              style={{ fontFamily: Fonts.Unica77LLRegular, fontSize: 18, marginLeft: 20 }}
-            >
-              {date}
-            </Text>
+          keyExtractor={(item) => item.internalID}
+          stickySectionHeadersEnabled
```
Comment (olerichter00) [inline src/lib/Scenes/AuctionResultsForYou/AuctionResultsForYou.tsx:69]: This enables the sticky section header behaviour for Android - it's disabled by default 💡

File: src/lib/Scenes/AuctionResultsForYou/AuctionResultsForYou.tsx [hunk 3]
```diff
@@ -51,42 +53,48 @@ export const AuctionResultsForYou: React.FC<Props> = ({ me, relay }) => {
...
+          renderSectionHeader={({ section: { sectionTitle } }) => (
+            <Flex bg="white" mx="2">
```
Comment (olerichter00) [inline src/lib/Scenes/AuctionResultsForYou/AuctionResultsForYou.tsx:71]: **Note:** We want a solid background for the header: `bg="white"`.

File: src/lib/Scenes/AuctionResultsForYou/AuctionResultsForYou.tsx [hunk 4]
```diff
@@ -51,42 +53,48 @@ export const AuctionResultsForYou: React.FC<Props> = ({ me, relay }) => {
...
+        <Flex>
+          <Text fontSize={14} lineHeight={21} textAlign="left" color="black60" mx={20} my={17}>
+            The latest auction results for the {""}
+            <LinkText onPress={() => navigate("/favorites", { passProps: { initialTab: Tab.artists } })}>
+              artists you follow
+            </LinkText>
+            . You can also look up more auction results on the insights tab on any artist’s page.
+          </Text>
+        </Flex>
         <SectionList
```
Comment (olerichter00) [inline src/lib/Scenes/AuctionResultsForYou/AuctionResultsForYou.tsx:65]: **Question:** I'm wondering if we should extract this `FlatList` into its own component to reuse it. Right now we don't use section lists (with sticky header) anywhere else but when we do it should look the same.

File: android/app/src/main/AndroidManifest.xml [hunk 5]
```diff
@@ -76,6 +76,7 @@
                 <data android:pathPrefix="/artworks" />
                 <data android:pathPrefix="/auction-faq" />
                 <data android:pathPrefix="/auction-registration" />
+                <data android:pathPrefix="/auction-results-for-you" />
```
Comment (MounirDhahri) [inline android/app/src/main/AndroidManifest.xml:79]: 💫

File: src/lib/Scenes/AuctionResultsForYou/AuctionResultsForYou.tsx (hunk 4 above)
Comment (MounirDhahri) [reply src/lib/Scenes/AuctionResultsForYou/AuctionResultsForYou.tsx:65]: this looks good to me this way. No need to extract it for now I guess until we need to use it in the future

Comment (MounirDhahri) [reply src/lib/Scenes/AuctionResultsForYou/AuctionResultsForYou.tsx:65]: Thanks for bringing that up though

File: src/lib/Scenes/AuctionResultsForYou/AuctionResultsForYou.tsx (hunk 2 above)
Comment (MounirDhahri) [reply src/lib/Scenes/AuctionResultsForYou/AuctionResultsForYou.tsx:69]: 👏

**Review by MounirDhahri — APPROVED** [review]

**Comment by MounirDhahri** [comment]
Thanks for making these changes and sorry for the late review, I will fix the merge conflicts! 🎉

**Comment by MounirDhahri** [comment]
#squashongreen

**Pull request merged**